content/sensu-go/5.17/guides/secrets-management.md

```diff
@@ -0,0 +1,289 @@
+---
+title: "Use secrets management in Sensu"
+linkTitle: "Use Secrets Management"
+description: "Sensu's secrets management allows you to avoid exposing secrets in your Sensu configuration. In this guide, you'll learn how to use Env or HashiCorp Vault as your external secrets management provider and refer to external secrets in your Sensu configuration."
+weight: 175
+version: "5.17"
+product: "Sensu Go"
+platformContent: False
+menu: 
+  sensu-go-5.17:
+    parent: guides
+---
+
+- [Use Env for secrets management](#use-env-for-secrets-management)
+  - [Create your backend environment variable](#create-your-backend-environment-variable)
+  - [Create your secret](#create-your-secret)
+- [Use HashiCorp Vault for secrets management](#use-hashicorp-vault-for-secrets-management)
+  - [Retrieve your Vault root token](#retrieve-your-vault-root-token)
+  - [Create your Vault secrets provider](#create-your-vault-secrets-provider)
+  - [Create your secret](#create-your-secret)
+- [Add a handler](#add-a-handler)
+  - [Register the PagerDuty handler asset](#register-the-pagerduty-handler-asset)
+  - [Add your secret to the handler spec](#add-your-secret-to-the-handler-spec)
+- [Next steps](#next-steps)
+
+**COMMERCIAL FEATURE**: Access the Env and VaultProvider secrets provider datatypes in the packaged Sensu Go distribution.
+For more information, see [Get started with commercial features][20].
+
+Sensu's secrets management allows you to avoid exposing secrets in your Sensu configuration.
+In this guide, you'll learn how to use Sensu's built-in secrets provider, `Env`, and [HashiCorp Vault][1] as your external [secrets provider][2] and authenticate without exposing secrets like usernames, passwords, and access keys.
+
+To follow this guide, you’ll need to [install the Sensu backend][5], have at least one [Sensu agent][11] running, and [install and configure sensuctl][7].
+
+Secrets are configured via [`secrets` resources][8].
+A secret resource definition refers to the secrets provider (`Env` or `VaultProvider`) and an ID (the named secret to fetch from the secrets provider).
+
+Your backend will execute a PagerDuty handler that requires your secret.
+The Sensu backend will transmit a request over its secure transport (TLS-encrypted websockets) to your Sensu agent to execute your handler.
+
+**Is this true for Vault? Even though we're using the no-TLS development server?**
+
+Your Sensu handler will include secrets that are exposed to Sensu services at runtime as environment variables.
+The `secrets` scope of the request payload and the environment variables are automatically redacted from all Sensu service logs and dashboards to prevent secret leakage.
```

Comment:
```suggestion
Sensu automatically redacts the `secrets` scope of the request payload and the environment variable from all Sensu service logs and dashboards to prevent secret leakage.
```